### ISSUE TITLE
Bug: When a Word Has a Hyphen in It the Second Word is not Capitalized when Capitalize Each Word is Used

### ISSUE THREAD

**Issue by pjkaufman** [platers/obsidian-linter#657]
## Describe the Bug

When a compound word like "Floating-point" is in a header it does not capitalize point when it is set to capitalize each word in the header.

## How to Reproduce

Run Linter on 

``` markdown
# Floating-point Types
```

## Expected Behavior

Expected output if applicable:

``` markdown
# Floating-Point Types
```

## Device

- [ ] Desktop
- [ ] Mobile

## Additional Context

This was reported by `@pinto` on Obsidian discord channel.

**Comment by pjkaufman** [comment]
This is not a bug. This is actually a feature request since there is no set standard on this. You can choose to capitalize any word after the hyphen or leave it as is.

See https://english.stackexchange.com/questions/460/do-you-capitalize-both-parts-of-a-hyphenated-word-in-a-title
